### PR TITLE
feat: token cache

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

closes: #1064

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by adamewozniak** [comment]
Not super sure we need this imo, would cacheing the leverage token registry reduce load by that much?

**Comment by robert-zaremba** [comment]
This is used in every tx method

**Comment by robert-zaremba** [comment]
Also this is a safe cache.

File: x/leverage/keeper/keeper.go [hunk 1]
```diff
@@ -28,19 +31,25 @@ func NewKeeper(
...
 		paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
 	}
 
-	return Keeper{
-		cdc:          cdc,
-		storeKey:     storeKey,
-		paramSpace:   paramSpace,
-		bankKeeper:   bk,
-		oracleKeeper: ok,
+	tokenRegCache, err := simplelru.NewLRU(100, nil)
```
Comment (adamewozniak) [inline x/leverage/keeper/keeper.go:40]: This argument should probably be a  descriptive constant

**Review by adamewozniak — APPROVED** [review]
utACK

**Comment by toteki** [comment]
Unit tests right now are getting a lot of this 

```
--- FAIL: TestSimTestSuite (0.04s)
    --- FAIL: TestSimTestSuite/TestSimulateMsgBorrowAsset (0.01s)
        suite.go:77: test panicked: runtime error: invalid memory address or nil pointer dereference
            goroutine 13 [running]:
            runtime/debug.Stack()
            	/opt/hostedtoolcache/go/1.18.3/x64/src/runtime/debug/stack.go:24 +0x65
            github.com/stretchr/testify/suite.failOnPanic(0xc0000c36c0, {0x19b1e20, 0x322f530})
            	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.5/suite/suite.go:77 +0x3b
            github.com/stretchr/testify/suite.Run.func1.1()
            	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.5/suite/suite.go:161 +0x252
            panic({0x19b1e20, 0x322f530})
            	/opt/hostedtoolcache/go/1.18.3/x64/src/runtime/panic.go:838 +0x[207](https://github.com/umee-network/umee/runs/7083806789?check_suite_focus=true#step:6:208)
            github.com/umee-network/umee/v2/x/leverage/keeper.Keeper.SetTokenSettings({{0x24fd788, 0xc0010562b0}, {0x24db420, 0xc00108c6e0}, {{0x24fb230, 0xc0010562b0}, 0xc00011f410, {0x24db420, 0xc00108c650}, {0x24db470, ...}, ...}, ...}, ...)
```
Is there something in `SetTokenSetting` that needs to be initialized?

File: x/leverage/keeper/keeper.go (hunk 1 above)
Comment (robert-zaremba) [reply x/leverage/keeper/keeper.go:40]: yes, good idea, adding it.

**Pull request merged**